### DISCUSSION TITLE
Autokeras: No module named 'tensorflow.keras.layers.experimental'

### DISCUSSION THREAD

**Discussion by Aya121298** [keras-team/autokeras#1905] [Q&A]
Everytime I try to use the import autokeras as ak function even though I have installed autokeras and tensorflow but it is always giving me: 'No module named 'tensorflow.keras.layers.experimental'

`!pip install autokeras`
`import tensorflow as tf`
`import autokeras as ak`

ModuleNotFoundError                       Traceback (most recent call last)
Cell In[7], line 1
----> 1 import autokeras as ak

File [~\AppData\Local\Programs\Python\Python311\Lib\site-packages\autokeras\__init__.py:1](http://localhost:8888/lab/tree/~/AppData/Local/Programs/Python/Python311/Lib/site-packages/autokeras/__init__.py#line=0)
----> 1 from autokeras.auto_model import AutoModel
      2 from autokeras.engine.block import Block
      3 from autokeras.engine.head import Head

File [~\AppData\Local\Programs\Python\Python311\Lib\site-packages\autokeras\auto_model.py:11](http://localhost:8888/lab/tree/~/AppData/Local/Programs/Python/Python311/Lib/site-packages/autokeras/auto_model.py#line=10)
      8 import tensorflow as tf
      9 from tensorflow.python.util import nest
---> 11 from autokeras import graph as graph_module
     12 from autokeras import hypermodels
     13 from autokeras import nodes as input_module

File [~\AppData\Local\Programs\Python\Python311\Lib\site-packages\autokeras\graph.py:5](http://localhost:8888/lab/tree/~/AppData/Local/Programs/Python/Python311/Lib/site-packages/autokeras/graph.py#line=4)
      2 import tensorflow as tf
      3 from tensorflow.python.util import nest
----> 5 from autokeras import hypermodels
      6 from autokeras import nodes
      7 from autokeras.engine import head as head_module

File [~\AppData\Local\Programs\Python\Python311\Lib\site-packages\autokeras\hypermodels\__init__.py:7](http://localhost:8888/lab/tree/~/AppData/Local/Programs/Python/Python311/Lib/site-packages/autokeras/hypermodels/__init__.py#line=6)
      5 from autokeras.hypermodels.basic import RNNBlock
      6 from autokeras.hypermodels.basic import XceptionBlock
----> 7 from autokeras.hypermodels.heads import ClassificationHead
      8 from autokeras.hypermodels.heads import RegressionHead
      9 from autokeras.hypermodels.preprocessing import CategoricalToNumerical

File [~\AppData\Local\Programs\Python\Python311\Lib\site-packages\autokeras\hypermodels\heads.py:5](http://localhost:8888/lab/tree/~/AppData/Local/Programs/Python/Python311/Lib/site-packages/autokeras/hypermodels/heads.py#line=4)
      2 from tensorflow.python.util import nest
      4 from autokeras import adapters
----> 5 from autokeras import keras_layers
      6 from autokeras import utils
      7 from autokeras.engine import head as head_module

File [~\AppData\Local\Programs\Python\Python311\Lib\site-packages\autokeras\keras_layers.py:7](http://localhost:8888/lab/tree/~/AppData/Local/Programs/Python/Python311/Lib/site-packages/autokeras/keras_layers.py#line=6)
      5 import tensorflow as tf
      6 from tensorflow.keras import backend as K
----> 7 from tensorflow.keras.layers.experimental import preprocessing
      8 from tensorflow.python.util import nest
     10 CombinerPreprocessingLayer = inspect.getmro(preprocessing.Normalization)[1]

ModuleNotFoundError: No module named 'tensorflow.keras.layers.experimental'

**Comment by SarahHamer** [comment]
Same issue. No solve yet though, sorry

**Comment by sohang3112** [comment]
@Aya121298 I'm having the same issue - did you find any solution to this??
